cpp/src/slice2swift/Gen.cpp

```diff
@@ -267,25 +267,13 @@ Gen::TypesVisitor::visitClassDefStart(const ClassDefPtr& p)
         out << eb;
     }
 
-    //
     // For each Value class we generate an extension method in ClassResolver
-    //
-    ostringstream factory;
-    factory << prefix;
-    vector<string> parts = splitScopedName(p->scoped());
-    for (auto it = parts.begin(); it != parts.end();)
-    {
-        factory << (*it);
-        if (++it != parts.end())
-        {
-            factory << "_";
-        }
-    }
+    const string factoryName = removeEscaping(prefix + p->mappedScoped("_").substr(1));
```

Comment:
This does not look correct.

The whole point of this "factory name" is to allow the Swift runtime to resolve a Slice type ID into a mapped class. So you can't use the mapped name in this factoryName!